[BugFix] Disable creating generated columns in lake and agg table (#26084)

srlch opened the pull request:
Problem:
1. Currently, generated column does not support lake table, but there is no restriction for creating it in lake table which is a bug.
2. agg table can be created generated column actually, but it is difficult to understand the semantic and it is no clearly scenes to use it. So we should disable it too.

Fixes #26084

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 19 (26.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterTableStatementAnalyzer.java|0|12|00.00%|[305, 306, 309, 310, 424, 425, 428, 429, 542, 543, 546, 547]|
|:large_blue_circle:|com/starrocks/server/OlapTableFactory.java|3|5|60.00%|[342, 343]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|2|2|100.00%|[]|

HangyuanLiu reviewed (commented):
move to create table analyzer

srlch committed:
[BugFix] Disables creating generated columns in lake and agg table (#26084)

Problem:
1. Currently, generated column does not support lake table, but there is no restriction for
   creating it in lake table which is a bug.
2. agg table can be created generated column actually, but it is difficult to understand the
   semantic and it is no clearly scenes to use it. So we should disable it too.

Signed-off-by: srlch <linzichao@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 16 (25.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/AlterTableStatementAnalyzer.java|0|12|00.00%|[305, 308, 309, 312, 423, 424, 427, 428, 541, 542, 545, 546]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|4|4|100.00%|[]|

wanpengfei-git commented:
@Mergifyio backport branch-3.1
